lib/web/servers.go

```diff
@@ -157,3 +158,53 @@ func (h *Handler) getDesktopHandle(w http.ResponseWriter, r *http.Request, p htt
 	// to see the desktop once in the UI, so just take the first one.
 	return ui.MakeDesktop(windowsDesktops[0]), nil
 }
+
+// desktopIsActive checks if a desktop has an active session and if so, returns a one time token that is needed
+// to start a session to said desktop.
+//
+// GET /v1/webapi/sites/:site/desktops/:desktopName/active
+//
+// Response body:
+//
+// {"active": bool}
+func (h *Handler) desktopIsActive(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	desktopName := p.ByName("desktopName")
+	trackers, err := h.auth.proxyClient.GetActiveSessionTrackers(r.Context())
```

Comment:
I'm a little worried this will scale poorly with the number of sessions. Can we somehow push the filter expression, i.e. `tracker.GetSessionKind() == types.WindowsDesktopSessionKind && tracker.GetState() == types.SessionState_SessionStateRunning && tracker.GetDesktopName() == desktopName` towards the backend where it can be used for efficient filtering? At the very least it should decrease the amount of traffic between the nodes.

If this is performance critical and the backend supports this, we could add an appropriate index to make this faster, too.

Reply:
I don't think we're at the point where indexing is necessary. Created a filter in 73f28a530aa3438402da31d1e5384de78050f069